### ISSUE TITLE
Void Bolt Timings are off

### ISSUE THREAD

**Issue by seanpeters86** [seanpeters86/bfa-shadow-priest#20]
Shot tested in-game vs. SimC timings of void bolt and noticed a pretty large difference.

<img width="600" alt="screen shot 2018-06-30 at 9 02 21 pm" src="https://user-images.githubusercontent.com/10604059/42130386-54698390-7ca9-11e8-8c9a-1366fd49fccd.png">

The times in the chart are the seconds between each void bolt. Either there is a problem with mind flay not respecting the `interrupt=1` in the APL, or Void Bolt isn't scaling properly with haste.

The problem doesn't really exist at all in the legion version of SimC, there's a chance our haste levels in legion just masked this problem, or something else changed between legion and bfa dev of SimC to break this.

[Full spreadsheet of data here](https://docs.google.com/spreadsheets/d/1H-dJkAuZFCbzwcULFKzeFtNTck9ghU5pi_4DK_u_HnU/edit?usp=sharing)

**Comment by N1gh7h4wk** [comment]
Did you make sure this isn't something to do with the latency on the beta?
Live thingy definitely looks like latency variance
bfa-dev looks weird if its no latency problem

**Comment by seanpeters86** [comment]
Confirm latency isn't part of the issue. Latency is equal on both, and if anything the sim is performing worse than the beta log, so latency wouldnt hurt in this case.

**Comment by seanpeters86** [comment]
N1gh7 suggested this new apl addition: https://github.com/simulationcraft/simc/pull/4344

Results: https://github.com/seanpeters86/bfa-shadow-priest/tree/master/apl

Need to redo tests to see if the issue is fixed

**Comment by seanpeters86** [comment]
Issue might still be active:

Shot has done some work here:
https://docs.google.com/spreadsheets/d/1BXq73NjojEEO8EE3Q2zGm5rIMS0vL2HpLQaVfdel9xc/edit?usp=sharing

**Comment by N1gh7h4wk** [comment]
![image](https://user-images.githubusercontent.com/22776166/42159300-92f13a7c-7df3-11e8-9099-ea252ae6d2cf.png)

**Comment by Ryeshot** [comment]
What I am trying to illustrate in the spreadsheet is that the SimC actor is casting a Shadow Word: Void at a point where 1) it is more optimal to continue casting Mind Flay and 2) it is not possible, due to Shadow's rotation's interactions with globals, to cast a SWV there _**unless**_ cooldowns are desynced through either improper flay clipping or incorrect scaling with haste on VB/SWV.

Cynth and I provided logs where we did its cast sequence and obviously didn't not cast VB on cooldown, due to the significantly different time between 4 and 5 VB from others. When we just channeled a full Flay, we cast VB roughly on cooldown.

![image](https://user-images.githubusercontent.com/31656086/42187206-3fb000bc-7e1d-11e8-9a99-23295b711448.png)

**Comment by Ryeshot** [comment]
It should also be noted for the SimC report, that the two long intervals (4.194 and 4.033sec) between VBs occur during Flay casts.

**Comment by seanpeters86** [comment]
Ansh noticed that SWV cd was not being adjusted properly, fix here: https://github.com/simulationcraft/simc/commit/61acd8f10098c4abdb719362596b64cba6381978

**Comment by djriff** [comment]
So I've looked into this issue thoroughly. SW: V recharges do benefit from haste buffs, including VF stacks. VB also benefits from this. After testing simc several times using various apls, calculating it manually in a sheet and testing it in game, I was able to match exaxtly what the sim was doing with no problems.

I think they issue you are running into is simc will cancel flay to cast VB, but won't actually update it's log data until a new spell is cast. This can lead to some confusion when analyzing the results.
